services/prometheus/prometheus.go

```diff
@@ -264,32 +268,56 @@ func (svc *Service) addScrapeConfigs(cfg *config.Config, q *reform.Querier, s *m
 	return nil
 }
 
+func (svc *Service) baseConfig(filename string, s models.MetricsResolutions) *config.Config {
+	cfg := &config.Config{
+		GlobalConfig: config.GlobalConfig{
+			// these values will be overwriten if they exit in the base config gfile
+			ScrapeInterval:     model.Duration(s.LR),
+			ScrapeTimeout:      scrapeTimeout(s.LR),
+			EvaluationInterval: model.Duration(s.LR),
+		},
+	}
+
+	buf, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return cfg
+	}
+
+	if err := yaml.Unmarshal(buf, cfg); err != nil {
+		svc.l.Warnf("Failed to parse base prometheus config %s: %s.", basePrometheusConfig, err)
+	}
+
+	return cfg
+}
+
+func inSlice(slice []string, val string) bool {
+	for _, elem := range slice {
+		if elem == val {
+			return true
+		}
+	}
+	return false
+}
+
 // marshalConfig marshals Prometheus configuration.
 func (svc *Service) marshalConfig() ([]byte, error) {
-	var cfg *config.Config
+	cfg := &config.Config{}
+
 	e := svc.db.InTransaction(func(tx *reform.TX) error {
 		settings, err := models.GetSettings(tx)
 		if err != nil {
 			return err
 		}
 		s := settings.MetricsResolutions
 
-		cfg = &config.Config{
-			GlobalConfig: config.GlobalConfig{
-				ScrapeInterval:     model.Duration(s.LR),
-				ScrapeTimeout:      scrapeTimeout(s.LR),
-				EvaluationInterval: model.Duration(s.LR),
-			},
-			RuleFiles: []string{
-				"/srv/prometheus/rules/*.rules.yml",
-			},
-			ScrapeConfigs: []*config.ScrapeConfig{
-				scrapeConfigForPrometheus(s.HR),
-				scrapeConfigForGrafana(s.MR),
-				scrapeConfigForPMMManaged(s.MR),
-				scrapeConfigForQANAPI2(s.MR),
-			},
+		cfg = svc.baseConfig(basePrometheusConfig, s)
+		if !inSlice(cfg.RuleFiles, defaultPrometheusRulesDir) {
```

Comment:
Why do that extra check? If this path is already in the base file, we should add it twice.